optuna/pruners/_wilcoxon.py

```diff
@@ -0,0 +1,160 @@
+            The larger this value is, the more aggressive pruning will be performed.
+            Defaults to 0.1.
+
+            .. note::
+                Contrary to the usual statistical wisdom, this pruner repeatedly
+                performs statistical tests between the current trial and the
+                current best trial with increasing samples.
+                Please expect around ~2x probability of falsely pruning
+                a good trial, compared to the usual false positive rate of
+                performing the statistical test only once.
+
+        n_startup_steps:
+            The number of steps before which no trials are pruned.
+            Defaults to 0 (pruning kicks in from the very first step).
+    """  # NOQA: E501
+
+    def __init__(
+        self,
+        *,
+        p_threshold: float = 0.1,
+        n_startup_steps: int = 0,
+    ) -> None:
+        if n_startup_steps < 0:
+            raise ValueError(f"n_startup_steps must be nonnegative but got {n_startup_steps}.")
+        if not 0.0 <= p_threshold <= 1.0:
+            raise ValueError(f"p_threshold must be between 0 and 1 but got {p_threshold}.")
+
+        self._n_startup_steps = n_startup_steps
+        self._p_threshold = p_threshold
+
+    def prune(self, study: "optuna.study.Study", trial: FrozenTrial) -> bool:
+        # Import scipy in the method because it is an optional dependency.
+        import scipy.stats as ss
+
+        if len(trial.intermediate_values) == 0:
+            return False
+
+        steps, step_values = np.array(list(trial.intermediate_values.items())).T
+
+        if np.any(np.isnan(step_values)):
+            warnings.warn(
+                f"The intermediate values of the current trial (trial {trial.number}) "
+                f"contain NaNs. WilcoxonPruner will not prune this trial."
+            )
+            return False
+
+        try:
+            best_trial = study.best_trial
+        except ValueError:
+            return False
+
+        best_steps, best_step_values = np.array(list(best_trial.intermediate_values.items())).T
+
+        if np.any(np.isnan(best_step_values)):
+            warnings.warn(
+                f"The intermediate values of the best trial (trial {best_trial.number}) "
+                f"contain NaNs. WilcoxonPruner will not prune the current trial."
+            )
+            return False
+
+        _, idx1, idx2 = np.intersect1d(steps, best_steps, return_indices=True)
+
+        if len(idx1) < len(step_values):
+            warnings.warn(
+                "WilcoxonPruner finds steps existing in the current trial "
+                "but does not exist in the best trial. "
+                "Those values are ignored."
+            )
+
+        diff_values = step_values[idx1] - best_step_values[idx2]
+        # Overwrite (inf - inf) or (-inf) - (-inf) with 0.
+        diff_values[step_values[idx1] == best_step_values[idx2]] = 0
+
+        if len(diff_values) < self._n_startup_steps:
```

Comment:
```
n_startup_steps:
            The number of steps before which no trials are pruned.
```
Based on the documentation, I feel whether `len(steps)` or `len(diff_values)` is used for this condition is ambiguous.